sql/src/main/java/org/opensearch/sql/sql/SQLService.java

```diff
@@ -75,6 +91,83 @@ private AbstractPlan plan(
                     .isExplain(request.isExplainRequest())
                     .build()));
 
+    // There is no full support for JSON format yet for in memory operations, aliases, literals, and casts
+    // Aggregation has differences with legacy results
+    if (request.format().getFormatName().equals("json")) {
+      class VisitorContext {
+        @Getter
+        @Setter
+        boolean isJSONSupported = true;
+      }
+
+      class JsonSupportedVisitor extends AbstractNodeVisitor<Boolean, VisitorContext> {
```

Comment:
Maybe move to another file?

Reply:
Thanks, moved in f6db3453e5b8aabdf10a6ba12016e311d0835c6b